hibernate-core/src/main/java/org/hibernate/query/sqm/sql/internal/EntityValuedPathInterpretation.java

```diff
@@ -351,38 +356,51 @@ public static <T> EntityValuedPathInterpretation<T> from(
 		);
 	}
 
-	private static boolean isSelected(TableGroup tableGroup, NavigablePath path, SqmQuerySpec<?> sqmQuerySpec) {
+	private static boolean isSelected(
+			TableGroup tableGroup,
+			NavigablePath path,
+			SqmQuerySpec<?> sqmQuerySpec,
+			boolean isTopLevel) {
 		// If the table group is selected (initialized), check if the entity valued
 		// navigable path or any child path appears in the select clause
-		return tableGroup.isInitialized() && selectClauseContains( path, sqmQuerySpec );
+		return tableGroup.isInitialized() && selectClauseContains(
+				path,
+				isTopLevel ? null : tableGroup.getNavigablePath(),
+				sqmQuerySpec
+		);
 	}
 
-	private static boolean selectClauseContains(NavigablePath path, SqmQuerySpec<?> sqmQuerySpec) {
+	private static boolean selectClauseContains(
+			NavigablePath path,
+			NavigablePath tableGroupPath,
+			SqmQuerySpec<?> sqmQuerySpec) {
 		final List<SqmSelection<?>> selections = sqmQuerySpec.getSelectClause() == null
 				? Collections.emptyList()
 				: sqmQuerySpec.getSelectClause().getSelections();
 		for ( SqmSelection<?> selection : selections ) {
-			if ( selectionContains( selection.getSelectableNode(), path ) ) {
+			if ( selectionContains( selection.getSelectableNode(), path, tableGroupPath ) ) {
 				return true;
 			}
 		}
 		return false;
 	}
 
-	private static boolean selectionContains(Selection<?> selection, NavigablePath path) {
-		if ( selection instanceof SqmPath && path.isParentOrEqual( ( (SqmPath<?>) selection ).getNavigablePath() ) ) {
-			return true;
+	private static boolean selectionContains(Selection<?> selection, NavigablePath path, NavigablePath tableGroupPath) {
+		if ( selection instanceof SqmPath<?> ) {
+			final SqmPath<?> sqmPath = (SqmPath<?>) selection;
+			return ( tableGroupPath == null || sqmPath.getLhs().getNavigablePath().equals( tableGroupPath ) )
```

Comment:
A comment here that explains why this condition must hold would be nice :)

Reply:
I've added a comment, let me know if that works.